data_model/src/account.rs

```diff
@@ -345,7 +314,148 @@ impl FromStr for AccountId {
+    pub fn all_account_signatures() -> Self {
+        Self::AllAccountSignaturesAnd(ConstVec::new_empty())
+    }
+
+    /// Checks whether the transaction contains all the signatures required by the `SignatureCheckCondition`.
+    pub fn check(
+        &self,
+        account_signatories: &btree_set::BTreeSet<PublicKey>,
+        transaction_signatories: &btree_set::BTreeSet<PublicKey>,
+    ) -> MustUse<bool> {
+        let result = match &self {
+            SignatureCheckCondition::AnyAccountSignatureOr(additional_allowed_signatures) => {
+                account_signatories
+                    .iter()
+                    .chain(additional_allowed_signatures.as_ref())
+                    .any(|allowed| transaction_signatories.contains(allowed))
+            }
+            SignatureCheckCondition::AllAccountSignaturesAnd(additional_required_signatures) => {
+                account_signatories
+                    .iter()
+                    .chain(additional_required_signatures.as_ref())
+                    .all(|required_signature| transaction_signatories.contains(required_signature))
+            }
+        };
+
+        MustUse::new(result)
+    }
+}
+
 /// The prelude re-exports most commonly used traits, structs and macros from this crate.
 pub mod prelude {
     pub use super::{Account, AccountId, SignatureCheckCondition};
 }
+
+#[cfg(test)]
+mod tests {
+    use iroha_crypto::{KeyPair, PublicKey};
+
+    use super::SignatureCheckCondition;
+
+    fn make_key() -> PublicKey {
+        KeyPair::generate().unwrap().public_key().clone()
+    }
+
+    fn check_signature_check_condition(
+        condition: &SignatureCheckCondition,
+        account_signatories: &[&PublicKey],
+        tx_signatories: &[&PublicKey],
+        result: bool,
+    ) {
+        let account_signatories = account_signatories.iter().copied().cloned().collect();
+        let tx_signatories = tx_signatories.iter().copied().cloned().collect();
+
+        assert_eq!(
+            condition.check(&account_signatories, &tx_signatories,).0,
+            result
+        );
+    }
+
+    #[test]
+    fn signature_check_condition_default() {
+        let key1 = make_key();
+        let key2 = make_key();
+        let key3 = make_key();
+        let condition = SignatureCheckCondition::default();
+
+        check_signature_check_condition(&condition, &[], &[], false);
+        check_signature_check_condition(&condition, &[&key1], &[], false);
+        check_signature_check_condition(&condition, &[], &[&key1], false);
+        check_signature_check_condition(&condition, &[&key1], &[&key1], true);
+        check_signature_check_condition(&condition, &[&key1], &[&key2], false);
+        check_signature_check_condition(&condition, &[&key1, &key2, &key3], &[&key1], true);
+        check_signature_check_condition(&condition, &[&key1, &key2, &key3], &[&key2], true);
+        check_signature_check_condition(&condition, &[&key1, &key2, &key3], &[&key3], true);
+    }
```

Comment:
Hmmm, I am not proficient in that topic, but couldn't it be covered with `proptest`s?

Reply:
I think it __might__ be useful.. Though, tbh, for such a simple implementation they might be overkill.

Also, the (possibly) unexpected behaviour of `All` mode with the empty set of account keys would be hard to discover (even though it's probably not possible to make an account w/o keys :sweat_smile:)